components/places/src/bookmark_sync/incoming.rs

```diff
@@ -121,31 +139,49 @@ impl<'a> IncomingApplicator<'a> {
                                  WHERE guid = :guid),
                                 (SELECT id FROM moz_tags
                                  WHERE tag = :tag))",
-                        &[(":guid", &b.record_id.as_guid().as_str()), (":tag", t)],
+                        &[(":guid", &record_id.as_guid().as_str()), (":tag", t)],
                     )?;
                 }
             };
         }
         Ok(())
     }
 
-    fn store_incoming_folder(&self, modified: ServerTimestamp, f: FolderRecord) -> Result<()> {
+    fn store_incoming_folder(&self, modified: ServerTimestamp, f: &JsonValue) -> Result<()> {
+        let mut validity = SyncedBookmarkValidity::Valid;
+
+        let record_id = unpack_id("id", f, &mut validity);
+        let parent_record_id = unpack_optional_id("parentid", f, &mut validity);
+        let date_added = unpack_optional_i64("dateAdded", f, &mut validity);
+        let title = unpack_optional_string("title", f, &mut validity);
+
+        let mut children: Vec<BookmarkRecordId> = vec![];
+        if let Some(array) = f["children"].as_array() {
+            for v in array {
+                if v.is_string() {
```

Comment:
If a child GUID isn't a string, let's fail the sync with an error here. If that happens, things have _really_ gone wrong, and I think we'd be better off bailing (and seeing the error in telemetry) than trying to apply a sketchy structure.

Reply:
I have added telemetry for a child GUID failure. Please let me know if the way I use telemetry is wrong.

Reply:
No worries! Reverted!